lib/dialects/sqlite/query-generator.js

```diff
@@ -79,8 +82,139 @@ const QueryGenerator = {
+      if (options.model.rawAttributes[key.split('.')[0]] && options.model.rawAttributes[key.split('.')[0]].type instanceof DataTypes.JSON) {
+        field = options.model.rawAttributes[key.split('.')[0]];
+        fieldType = field.type;
+        const tmp = value;
+        value = {};
+
+        _.set(value, key.split('.').slice(1), tmp);
+        key = field.field || key.split('.')[0];
+      }
+    }
+
+    if (_.isPlainObject(value) && fieldType instanceof DataTypes.JSON && options.json !== false) {
+      const items = [];
+      const traverse = (prop, item, path) => {
+        const where = {};
+        let cast;
+
+        if (path[path.length - 1].indexOf('::') > -1) {
+          const tmp = path[path.length - 1].split('::');
+          cast = tmp[1];
+          path[path.length - 1] = tmp[0];
+        }
+
+        let baseKey = this.quoteIdentifier(key);
+
+        if (options.prefix) {
+          if (options.prefix instanceof Utils.Literal) {
+            baseKey = this.handleSequelizeMethod(options.prefix)+'.'+baseKey;
+          } else {
+            baseKey = this.quoteTable(options.prefix)+'.'+baseKey;
+          }
+        }
 
+        baseKey = `json_extract(${baseKey}, '$.${path.join('.')}')`;
+
+        const castKey = item => {
+          let key = baseKey;
+
+          if (!cast) {
+            if (typeof item === 'number') {
+              cast = 'double precision';
+            } else if (item instanceof Date) {
+              cast = 'datetime';
+            } else if (typeof item === 'boolean') {
+              cast = 'boolean';
+            }
+          }
+
+          if (cast) {
+            return this.handleSequelizeMethod(new Utils.Cast(new Utils.Literal(key), cast));
+          }
+
+          return key;
+        };
+
+        if (_.isPlainObject(item)) {
+          _.forOwn(item, (item, prop) => {
+            if (prop.indexOf('$') === 0) {
+              where[prop] = item;
+              const key = castKey(item);
+
+              items.push(this.whereItemQuery(new Utils.Literal(key), where/*, _.pick(options, 'prefix')*/));
+            } else {
+              traverse(prop, item, path.concat([prop]));
+            }
+          });
+        } else {
+          where.$eq = item;
+          const key = castKey(item);
+
+          items.push(this.whereItemQuery(new Utils.Literal(key), where/*, _.pick(options, 'prefix')*/));
+        }
+      };
+
+      _.forOwn(value, (item, prop) => {
+        if (prop.indexOf('$') === 0) {
+          const where = {};
+          where[prop] = item;
+          items.push(this.whereItemQuery(key, where, _.assign({}, options, {json: false})));
+          return;
+        }
+
+        traverse(prop, item, [prop]);
+      });
+
+      const result = items.join(' AND ');
+      return items.length > 1 ? '('+result+')' : result;
+    } else {
+      return AbstractQueryGenerator.whereItemQuery.call(this, key, value, options);
```

Comment:
Calling `super` here will trigger `MySQLQueryGen` (as SQLITE inherits it), but `super.whereItemQuery` is not present in MySQLQueryGen (Although SQLite -> MySQL -> Abstract is true) but we should refer to AbstractQueryGen as we want to switch from MySQLQueryGen in future.